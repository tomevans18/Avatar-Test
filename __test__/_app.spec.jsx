import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../pages/_app';

const mockRouterContext = jest.fn();

jest.mock('next/router', () => ({
  makePublicRouterInstance: () => {},
  withRouter: Component => props => <Component {...mockRouterContext()} {...props} />,
}));

describe('App', () => {
  beforeEach(() => {
    mockRouterContext.mockReturnValue({
      router: {
        pathname: '/',
      },
    });
  });

  afterEach(() => {
    mockRouterContext.mockReset();
  });

  const MockComponent = () => <div>TEST</div>;

  it('should render as expected with no passed component', () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render as expected with passed component', async () => {
    const props = await App.getInitialProps({
      Component: MockComponent,
    });
    const wrapper = mount(<App {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render only reset styles', () => {
    MockComponent.noGlobalStyles = true;
    const wrapper = shallow(<App Component={MockComponent} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with all styles', () => {
    MockComponent.noGlobalStyles = false;
    const wrapper = shallow(<App Component={MockComponent} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass down to component getInitialProps', async () => {
    MockComponent.getInitialProps = jest.fn();
    const props = await App.getInitialProps({
      Component: MockComponent,
    });
    const wrapper = mount(<App {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(MockComponent.getInitialProps).toHaveBeenCalledTimes(1);
  });

  it('should render as expected with no layout passed with component', () => {
    const wrapper = shallow(<App Component={MockComponent} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render as expected with layout passed with component', () => {
    MockComponent.layout = ({ Component, ...passedProps } = {}) => (
      <section>
        <Component {...passedProps} />
      </section>
    );
    const wrapper = shallow(<App Component={MockComponent} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
