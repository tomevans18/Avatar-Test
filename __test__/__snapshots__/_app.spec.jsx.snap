// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should pass down to component getInitialProps 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 *:focus {
  outline: 0;
}

<MyApp
  pageProps={Object {}}
  user={
    Object {
      "isLoggedIn": true,
      "userName": "testUser",
    }
  }
>
  <Container>
    <Head>
      <SideEffect
        handleStateChange={null}
        isAmp={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <AppProviders
      user={
        Object {
          "isLoggedIn": true,
          "userName": "testUser",
        }
      }
    >
      <ThemeProvider
        theme={
          Object {
            "breakpoint": Object {
              "extraLarge": 1224,
              "large": 1025,
              "medium": 769,
              "small": 451,
            },
            "color": Object {
              "black": "#1e1e1e",
              "grey": "#ccc",
              "white": "#ffffff",
            },
            "component": Object {
              "spacing": "1.25em",
            },
          }
        }
      >
        <ResizeAnimationHandler>
          <styled.div
            resizing={false}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c0",
                    "rules": Array [
                      "
  display: flex;
  flex-direction: column;
  flex: 1;

  ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              resizing={false}
            >
              <div
                className="c0"
              >
                <Index>
                  <styled.div
                    tabbing={false}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bwzfXH",
                            "isStatic": false,
                            "lastClassName": "c1",
                            "rules": Array [
                              "
  display: flex;
  flex-direction: column;
  flex: 1;

  *:focus {
    outline: ",
                              [Function],
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bwzfXH",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      tabbing={false}
                    >
                      <div
                        className="c1"
                      />
                    </StyledComponent>
                  </styled.div>
                </Index>
              </div>
            </StyledComponent>
          </styled.div>
        </ResizeAnimationHandler>
      </ThemeProvider>
    </AppProviders>
  </Container>
</MyApp>
`;

exports[`App should render as expected with layout passed with component 1`] = `
<Container>
  <Head>
    <title>
      Avatar Test
    </title>
  </Head>
  <GlobalStyleComponent />
  <AppProviders
    protectedPage={false}
    user={
      Object {
        "avatar": null,
        "isLoggedIn": false,
        "protectedPage": false,
        "status": "Available",
        "userName": null,
      }
    }
  >
    <Index>
      <Component
        Component={[Function]}
      />
    </Index>
  </AppProviders>
</Container>
`;

exports[`App should render as expected with no layout passed with component 1`] = `
<Container>
  <Head>
    <title>
      Avatar Test
    </title>
  </Head>
  <GlobalStyleComponent />
  <AppProviders
    protectedPage={false}
    user={
      Object {
        "avatar": null,
        "isLoggedIn": false,
        "protectedPage": false,
        "status": "Available",
        "userName": null,
      }
    }
  >
    <Index>
      <Foundation
        user={
          Object {
            "avatar": null,
            "isLoggedIn": false,
            "protectedPage": false,
            "status": "Available",
            "userName": null,
          }
        }
      >
        <MockComponent />
      </Foundation>
    </Index>
  </AppProviders>
</Container>
`;

exports[`App should render as expected with no passed component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 *:focus {
  outline: 0;
}

<MyApp
  protectedPage={false}
  user={
    Object {
      "avatar": null,
      "isLoggedIn": false,
      "protectedPage": false,
      "status": "Available",
      "userName": null,
    }
  }
>
  <Container>
    <Head>
      <SideEffect
        handleStateChange={null}
        isAmp={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <AppProviders
      protectedPage={false}
      user={
        Object {
          "avatar": null,
          "isLoggedIn": false,
          "protectedPage": false,
          "status": "Available",
          "userName": null,
        }
      }
    >
      <ThemeProvider
        theme={
          Object {
            "breakpoint": Object {
              "extraLarge": 1224,
              "large": 1025,
              "medium": 769,
              "small": 451,
            },
            "color": Object {
              "available": "rgb(46, 204, 113)",
              "background": "rgb(53, 59, 72)",
              "black": "#1e1e1e",
              "blue": "#2d8bd8",
              "grey": "rgb(220, 221, 225)",
              "grey_light": "#eaeaea",
              "notReady": "rgb(251, 197, 49)",
              "unavailable": "rgb(194, 54, 22)",
              "white": "#ffffff",
            },
            "component": Object {
              "spacing": "1.25em",
            },
          }
        }
      >
        <UserProvider
          protectedPage={false}
          user={
            Object {
              "avatar": null,
              "isLoggedIn": false,
              "protectedPage": false,
              "status": "Available",
              "userName": null,
            }
          }
        >
          <ResizeAnimationHandler>
            <styled.div
              resizing={false}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  display: flex;
  flex-direction: column;
  flex: 1;

  ",
                        [Function],
                        ";
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                resizing={false}
              >
                <div
                  className="c0"
                >
                  <Index>
                    <styled.div
                      tabbing={false}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "c1",
                              "rules": Array [
                                "
  display: flex;
  flex-direction: column;
  flex: 1;

  *:focus {
    outline: ",
                                [Function],
                                ";
  }
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        tabbing={false}
                      >
                        <div
                          className="c1"
                        />
                      </StyledComponent>
                    </styled.div>
                  </Index>
                </div>
              </StyledComponent>
            </styled.div>
          </ResizeAnimationHandler>
        </UserProvider>
      </ThemeProvider>
    </AppProviders>
  </Container>
</MyApp>
`;

exports[`App should render as expected with passed component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 *:focus {
  outline: 0;
}

<MyApp
  user={
    Object {
      "isLoggedIn": true,
      "userName": "testUser",
    }
  }
>
  <Container>
    <Head>
      <SideEffect
        handleStateChange={null}
        isAmp={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <AppProviders
      user={
        Object {
          "isLoggedIn": true,
          "userName": "testUser",
        }
      }
    >
      <ThemeProvider
        theme={
          Object {
            "breakpoint": Object {
              "extraLarge": 1224,
              "large": 1025,
              "medium": 769,
              "small": 451,
            },
            "color": Object {
              "black": "#1e1e1e",
              "grey": "#ccc",
              "white": "#ffffff",
            },
            "component": Object {
              "spacing": "1.25em",
            },
          }
        }
      >
        <ResizeAnimationHandler>
          <styled.div
            resizing={false}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c0",
                    "rules": Array [
                      "
  display: flex;
  flex-direction: column;
  flex: 1;

  ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              resizing={false}
            >
              <div
                className="c0"
              >
                <Index>
                  <styled.div
                    tabbing={false}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bwzfXH",
                            "isStatic": false,
                            "lastClassName": "c1",
                            "rules": Array [
                              "
  display: flex;
  flex-direction: column;
  flex: 1;

  *:focus {
    outline: ",
                              [Function],
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bwzfXH",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      tabbing={false}
                    >
                      <div
                        className="c1"
                      />
                    </StyledComponent>
                  </styled.div>
                </Index>
              </div>
            </StyledComponent>
          </styled.div>
        </ResizeAnimationHandler>
      </ThemeProvider>
    </AppProviders>
  </Container>
</MyApp>
`;

exports[`App should render only reset styles 1`] = `
<Container>
  <Head>
    <title>
      Avatar Test
    </title>
  </Head>
  <GlobalStyleComponent />
  <AppProviders
    protectedPage={false}
    user={
      Object {
        "avatar": null,
        "isLoggedIn": false,
        "protectedPage": false,
        "status": "Available",
        "userName": null,
      }
    }
  >
    <Index>
      <Foundation
        user={
          Object {
            "avatar": null,
            "isLoggedIn": false,
            "protectedPage": false,
            "status": "Available",
            "userName": null,
          }
        }
      >
        <MockComponent />
      </Foundation>
    </Index>
  </AppProviders>
</Container>
`;

exports[`App should render with all styles 1`] = `
<Container>
  <Head>
    <title>
      Avatar Test
    </title>
  </Head>
  <GlobalStyleComponent />
  <AppProviders
    protectedPage={false}
    user={
      Object {
        "avatar": null,
        "isLoggedIn": false,
        "protectedPage": false,
        "status": "Available",
        "userName": null,
      }
    }
  >
    <Index>
      <Foundation
        user={
          Object {
            "avatar": null,
            "isLoggedIn": false,
            "protectedPage": false,
            "status": "Available",
            "userName": null,
          }
        }
      >
        <MockComponent />
      </Foundation>
    </Index>
  </AppProviders>
</Container>
`;
